build.go

```diff
@@ -112,7 +113,17 @@ func buildNode(k *Kong, v reflect.Value, typ NodeType, seenFlags map[string]bool
 	if err != nil {
 		return nil, err
 	}
+
+	MAIN:
 	for _, field := range fields {
+		for _, r := range k.ignoreFieldsRegex {
+			if r.Match([]byte(field.field.Name)) {
```

Comment:
Use `MatchString()` instead of casting.

Reply:
Please match against `v.Type().Name + "." + field.field.Name` so that it can match against the containing struct as well. It's fairly common in deeply nested CLI structures to repeat fields and being able to discriminate by the containing "command" will future proof this.

Reply:
Gotcha. Will do.